sale_order_type/models/account_move.py

```diff
@@ -0,0 +1,55 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# Copyright 2020 Tecnativa - Pedro M. Baeza
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    sale_type_id = fields.Many2one(
+        comodel_name="sale.order.type",
+        string="Sale Type",
+        compute="_compute_sale_type_id",
+        store=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        ondelete="restrict",
+        copy=True,
+    )
+
+    @api.depends("partner_id", "company_id")
+    def _compute_sale_type_id(self):
+        # If create invoice from sale order, sale type will not computed.
+        if not self.env.context.get("default_move_type", False) or self.env.context.get(
+            "active_model", False
+        ) in ["sale.order", "sale.advance.payment.inv"]:
+            return
+        self.sale_type_id = self.env["sale.order.type"]
+        for record in self:
+            if record.move_type not in ["out_invoice", "out_refund"]:
+                record.sale_type_id = self.env["sale.order.type"]
+                continue
+            else:
+                record.sale_type_id = record.sale_type_id
```

Comment:
I'm not sure of this. Maybe :

```suggestion
                record.sale_type_id = record._origin.sale_type_id
```

is better